src/main/fc/core.c

```diff
@@ -214,6 +214,61 @@ void resetArmingDisabled(void)
     lastArmingDisabledReason = 0;
 }
 
+#ifdef USE_ACC
+static bool accNeedsCalibration(void)
+{
+    if (sensors(SENSOR_ACC)) {
+
+        // Check to see if the ACC has already been calibrated
+        if (accelerometerConfig()->calibrationCompleted ||
+            accelerometerConfig()->accZero.values.roll != 0 ||
+            accelerometerConfig()->accZero.values.pitch != 0 ||
+            accelerometerConfig()->accZero.values.yaw != 0) {
```

Comment:
Not sure we should treat having a calibration value that is not null as a special case here - to me using only `calibrationCompleted` to determine if we have calibrated, and exposing it as a parameter would make more sense:
- it makes it obvious in a `diff` if ACC has been calibrated, even in the (unlikely) case of all calibration values being 0;
- it still works as intended even if a custom config is setting any calibration values to non-0.

Reply:
Sorry, forgot to get back to this.  The reasoning behind implementing the logic this way was to facilitate propagating the ACC calibration across builds through a `diff`. Since the ACC calibration is related to how the board is mounted in the frame rather than the firmware release, the `acc_calibration` parameter which is part of the `diff` can restore a user's calibration. For example I have this in all my `diff` templates that I apply as basic settings whenever upgrading firmware. So with the logic in this PR that will be enough to confirm that calibration is "intended". The `calibrationCompleted` is intended to only handle the (unlikely) special case of a calibration being performed but it turns out to be exactly all zeros. The problem with including `calibrationCompleted` in the `diff` is that the user could inappropriately propagate that setting across builds with a `diff` which would be a problem if the also neglected to propagate `acc_calibration` - effectively thwarting the checks. So in the end not a big deal, but that was the reasoning. Also the desire to not add another potentially confusing parameter that we don't want the users to manually set.

Reply:
I am still not a fan of using hidden config settings for this - how about change the reset function to reset `accZero` to `{ -1, -1, -1 }` and just check that none of them are negative to determine if calibration is needed?

Reply:
The problem is that just like all zeros `{ -1, -1, -1 }` is a valid possible calibration. The axis zero offset values are `int16` and negative offsets are valid and common.

Reply:
Erg. I'd only ever seen (small) positive values for acc calibration. How about we change one of the axes to `int32` and default it to -32769?

Reply:
I don't think that would be a good idea. The axis value range from the sensor is `int16`. If we defaulted to some out-of-range value as an indicator of lack of calibration then we'd also have to escape every use of the offsets to ensure we don't apply this false calibration to actual sensor data. To me this would be far more confusing in the code.

Reply:
And we couldn't change just one of the axes (at least easily). The `acc_calibration` parameter is a `MODE_ARRAY` type and it won't handle that. Also the zero offsets are an array of `int16` so we'd have to separate out to individual parameters and go to something like `acc_calibration_x`, `acc_calibration_y`, etc.

Reply:
Erg yes, no easy way to do this. To me, using a hidden persisted parameter that is confusing as well, plus it makes it hard to debug, especially since most accelerometers seem to calibrate close to the default values. Having an indicator that is coupled to the values (to avoid the problem of users accidentally keeping the indicator but not the calibration values) seems to be much better to me. How about adding a 4th value to the array to encode the calibration state in?

Reply:
> How about adding a 4th value to the array to encode the calibration state in?

The downside to that would be that again the `calibrationCompleted` could be propagated through the `diff` and applied to a new build. But if the concern is about exposing this value to indicate that a "hard calibration" took place from a support perspective, then another option would be to expose the `calibrationCompleted` value as part of the `status` output. Something like:
```
MCU F40X Clock=168MHz (PLLP-HSE), Vref=3.29V, Core temp=33degC
Stack size: 2048, Stack address: 0x1000fff0
Config size: 3781, Max available config: 16384
GYRO=MPU6000, ACC=MPU6000 (calibrated)
```

Reply:
> The downside to that would be that again the calibrationCompleted could be propagated through the diff and applied to a new build.

This is no different to your currently proposed solution, where any non-0 calibration data is propagated through to a new build and results in the calibration being recognised as complete. However, I don't think that there is an 'always right' solution to this - on one hand we **do want** the calibration to propagate through a `diff` when done as part of a firmware upgrade, to avoid users repeatedly having to calibrate, whereas we **do not want** this to happen when a `diff` from one flight controller is applied to another one. I think the first case is way more common, and thus outweighs the second.
Also, in my proposal the behaviour is at least consistent, and the calibration always propagates, whereas in the current solution the calibration only propagates most of the time, but in the case where an ACC is correctly calibrated at 0 it never propagates. Not sure this is desirable behaviour.

Reply:
Well ok, but even so we need to keep the logic that checks for any non-zero values in the array. Assuming we use the 4th element for a flag (0 = not calibrated, 1  = calibrated). If a user pastes in their current calibration from a saved diff that only had the 3 values like:
```
set acc_calibration = -20,13,4
```
Then the 4th element will get reset to 0 as that's the way `MODE_ARRAY` works. So checking for any of the 4 elements being non-zero will work, but can't just check for the 4th element to indicate that calibration is valid.

Reply:
I made the changes to include the `calibrationCompleted` flag as the 4th element of the `accZero` array. I did it as a separate commit because I think it ended up being a little ugly and to leave the option open to revert. The problem is that the structure used to store the calibration data is shared with the compass. So now that extra value is also available but unused for the compass (I didn't make the `compass_calibration` output the 4th value). I suppose we could always add similar calibration checks for the compass as some point so maybe not wasted. Since the structure was increased in size I had to increment the PG version for both `accelerometerConfig` and `compassConfig`.

So now the parameter will output like:
```
acc_calibration = -20,13,4,1
```
With the last value representing a boolean flag whether calibration was completed. Seems a bit ugly wasting a `int16` for just a flag, but it has to be since it's part of an array.